### ISSUE TITLE
Possibility to batch images per step?

### ISSUE THREAD

**Issue by abrahamezzeddine** [yzslab/gaussian-splatting-lightning#123]
Hello,

Is there a possibility to batch images per iteration to increase quality/improve convergence?

**Comment by yzslab** [comment]
No. However, a workaround is to use multi-GPU mode on a single GPU. For example, if you want to use `batch_size = 4`, you can start four training processes on a single GPU.

**Comment by abrahamezzeddine** [comment]
> No. However, a workaround is to use multi-GPU mode on a single GPU. For example, if you want to use `batch_size = 4`, you can start four training processes on a single GPU.

Ok, thanks. And that will make the model "converge" faster you mean?

In addition to that, when I use depth regularization + appearnace model, it quickly fills up my memory... I am using around 2400 images and fills up around 120GB of my RAM during training which seems quite excessive. With no depth regularization, it's normally around 15GB of RAM, max.

**Comment by yzslab** [comment]
> Ok, thanks. And that will make the model "converge" faster during each step?

Theoretically, but hyperparameters need to be tuned carefully too.



> fills up around 120GB of my RAM during training

What are the sizes of your images?

**Comment by abrahamezzeddine** [comment]
> > Ok, thanks. And that will make the model "converge" faster during each step?
> 
> Theoretically, but hyperparameters need to be tuned carefully too.
> 
> > fills up around 120GB of my RAM during training
> 
> What are the sizes of your images?

Not very large. Each image is around 650kB but the depth anything creates almost 15MB large files per image.

**Comment by yzslab** [comment]
> 15MB large files per image

Looks like your images are 1440P.


> With no depth regularization, it's normally around 15GB of RAM, max.

Are you sure? Theoretically, storing 2400 1440P RGB images in float32 requires about 100GB of memory. Storing in uint8 requires 25GB.
The depth maps should take about 33GB.

**Comment by yashj235** [comment]
@yzslab , I have a similar query. I used  `--data.train_max_num_images_to_cache 128` for a 1500 image scene when I was getting OOM errors. Will this impact performance (not speed vise, accuracy vise)?

**Comment by yzslab** [comment]
No, it only impacts the training speed.
